bootstrap_cfn/iam.py

```diff
@@ -29,7 +31,57 @@ def upload_ssl_certificate(self, ssl_config, stack_name):
         return True
 
     def delete_ssl_certificate(self, ssl_config, stack_name):
+        cert_list = self.conn_iam.list_server_certs()
+
         for cert_name in ssl_config.keys():
             cert_id = "{0}-{1}".format(cert_name, stack_name)
-            self.conn_iam.delete_server_cert(cert_id)
-        return True
+            # Check for cert in list before deleting
+            cert_list_metadata = cert_list['list_server_certificates_response']['list_server_certificates_result']['server_certificate_metadata_list']
+            found = False
+            for remote_cert in cert_list_metadata:
+                remote_cert_name = remote_cert['server_certificate_name']
+                if remote_cert_name == cert_id:
+                    logging.info("Deleting certificate '%s'.." % (remote_cert_name))
+                    self.conn_iam.delete_server_cert(cert_id)
+                    found = True
+                    break
+
+        if found:
+            logging.info("Deleting certificate '%s'.." % (remote_cert_name))
+        else:
+            logging.warn("Could not find certificate '%s'" % (remote_cert_name))
+
+        return found
+
+    def get_arn_for_cert(self, cert_name):
+        """
+        Use a certificates name to find the arn
+
+        Args:
+            cert_name (string): The name of the certification
+
+        Returns:
+            cert_arn (string): The certifications arn if found,
+                None type otherwise
+        """
+        cert_arn = None
+
+        # Collect all certificates into a list
+        certs = self.conn_iam.get_all_server_certs()
```

Comment:
Can you use `iam.get_server_certificate(cert_name)` instead? http://boto.readthedocs.org/en/latest/ref/iam.html#boto.iam.connection.IAMConnection.get_server_certificate